tests/native/test_models.py

```diff
@@ -198,3 +227,123 @@ def test_symbolic_mixed(self):
+        self._test_strcpy("abcdefghijklm\0")
+        self._test_strcpy("a\0", dst_len=5)
+
+    def test_concrete_empty(self):
+        self._test_strcpy("\0")
+        self._test_strcpy("\0", dst_len=10)
+
+    def test_symbolic(self):
+        BIN_PATH = os.path.join(os.path.dirname(__file__), "binaries", "sym_strcpy_test")
+        tmp_dir = tempfile.TemporaryDirectory(prefix="mcore_test_sym_")
+        m = Manticore(BIN_PATH, stdin_size=10, workspace_url=str(tmp_dir.name))
+
+        addr_of_strcpy = 0x400418
+
+        @m.hook(addr_of_strcpy)
+        def strlen_model(state):
+            state.invoke_model(strcpy)
+
+        m.run()
+        m.finalize()
+
+        # Approximate regexes for expected testcase output
+        # Example Match above each regex
+        # Manticore varies the hex output slightly per run
+        expected = [
+            # STDIN: b'\x00AAAAAAAAA'
+            "STDIN: b\\'\\\x00A.{8,32}\\'",
+            # STDIN: b'\xffA\x00\xff\xff\xff\xff\xff\xff\xff'
+            "STDIN: b\\'(\\\\x((?!(00))[0-9a-f]{2}))A\\\\x00(\\\\x([0-9a-f]{2})){7}\\'",
+            # STDIN: b'\xffA\xff\x00\xff\xff\xff\xff\xff\xff'
+            "STDIN: b\\'(\\\\x((?!(00))[0-9a-f]{2}))A(\\\\x((?!(00))[0-9a-f]{2}))\\\\x00(\\\\x([0-9a-f]{2})){6}\\'",
+            # STDIN: b'\xffA\xff\xff\x00\xff\xff\xff\xff\xff'
+            "STDIN: b\\'(\\\\x((?!(00))[0-9a-f]{2}))A(\\\\x((?!(00))[0-9a-f]{2})){2}\\\\x00(\\\\x([0-9a-f]{2})){5}\\'",
+            # STDIN: b'\x00\xbe\xbe\xbe\xbe\xbe\xbe\xbe\xbe\xbe'
+            "STDIN: b\\'\\\\x00(\\\\x([0-9a-f]{2})){9}\\'",
+            # STDIN: b'\xff\x00\xff\xff\xff\xff\xff\xff\xff\xff'
+            "STDIN: b\\'(\\\\x((?!(00))([0-9a-f]{2}))){1}\\\\x00(\\\\x([0-9a-f]{2})){8}\\'",
+            # STDIN: b'\xff\xff\x00\xff\xff\xff\xff\xff\xff\xff'
+            "STDIN: b\\'(\\\\x((?!(00))([0-9a-f]{2}))){2}\\\\x00(\\\\x([0-9a-f]{2})){7}\\'",
+            # STDIN: b'\xff\xff\xff\x00\xff\xff\xff\xff\xff\xff'
+            "STDIN: b\\'(\\\\x((?!(00))([0-9a-f]{2}))){3}\\\\x00(\\\\x([0-9a-f]{2})){6}\\'",
+            # STDIN: b'\xff\xff\xff\xff\x00\xff\xff\xff\xff\xff'
+            "STDIN: b\\'(\\\\x((?!(00))([0-9a-f]{2}))){4}\\\\x00(\\\\x([0-9a-f]{2})){5}\\'",
+            # STDIN: b'\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff'
+            "STDIN: b\\'(\\\\x((?!(00))([0-9a-f]{2}))){10}\\'",
+        ]
+
+        inputs = f"{str(m.workspace)}/test_*.input"
+
+        # Make a list of the generated input states
+        stdins = []
+        for inpt in glob(inputs):
+            with open(inpt) as f:
+                stdins.append(f.read())
+
+        # Check the number of input states matches the number of regexes
+        self.assertEqual(len(stdins), len(expected))
+
+        # Assert that every regex has a matching input
+        for e in expected:
+            match = False
+            for s in stdins:
+                if re.fullmatch(e, s) == None:
+                    match = True
```

Comment:
For efficiency sake, we should `break` after setting `match = True` to short-circuit after we find a match.

We could also remove the matched stdin string from `stdins` to cut down the complexity, but I'm not sure if the removal from the list would save us any time 🤷  Up to you if you want to do this part.

Reply:
Yea I thought about doing the list removal but I didn't think that would lower the time complexity because removal is O(n). So the break is probably better 😄